### DISCUSSION TITLE
v7.1.3

### DISCUSSION THREAD

**Discussion by kayhantolga** [betalgo/openai#342] [Announcements]
### 7.1.3
- This release was a bit late and took longer than expected due to a couple of reasons. The future was quite big, and I couldn't cover all possibilities. However, I believe I have covered most of the function definitions (with some details missing). Additionally, I added an option to build it manually. If you don't know what I mean, you don't need to worry. I plan to cover the rest of the function definition in the next release. Until then, you can discover this by playing in the playground or in the source code. This version also support using other libraries to export your function definition.
- We now have support for functions! Big cheers to @rzubek for completing most of this feature.
- Additionally, we have made bug fixes and improvements. Thanks to @choshinyoung, @yt3trees, @WeihanLi, @N0ker, and all the bug reporters. (Apologies if I missed any names. Please let me know if I missed your name and you have a commit.) 
### 7.1.2-beta
- Bugfix https://github.com/betalgo/openai/pull/302
- Added support for Function role https://github.com/betalgo/openai/issues/303
### 7.1.0-beta
- Function Calling: We're releasing this version to bring in a new feature that lets you call functions faster. But remember, this version might not be perfectly stable and we might change it a lot later. A big shout-out to @rzubek for helping us add this feature. Although I liked his work, I didn't have enough time to look into it thoroughly. Still, the tests I did showed it was working, so I decided to add his feature to our code. This lets everyone use it now. Even though I'm busy moving houses and didn't have much time, seeing @rzubek's help made things a lot easier for me.
- Support for New Models: This update also includes support for new models that OpenAI recently launched. I've also changed the naming style to match OpenAI's. Model names will no longer start with 'chat'; instead, they'll start with 'gpt_3_5' and so on.

## What's Changed
* Feature/plugins by @kayhantolga in https://github.com/betalgo/openai/pull/280
* Betalgo.OpenAI.Utilities by @kayhantolga in https://github.com/betalgo/openai/pull/281
* fix by @kayhantolga in https://github.com/betalgo/openai/pull/282
* Betalgo.OpenAI.Utilities by @kayhantolga in https://github.com/betalgo/openai/pull/283
* path fix by @kayhantolga in https://github.com/betalgo/openai/pull/284
* Updated libraries to latest versions. by @kayhantolga in https://github.com/betalgo/openai/pull/288
* v 7.0.1 by @kayhantolga in https://github.com/betalgo/openai/pull/289
* Added support for the new function calling capability in Chat Completions API by @rzubek in https://github.com/betalgo/openai/pull/300
* Update Readme.md by @N0ker in https://github.com/betalgo/openai/pull/297
* 7.1.0 -beta by @kayhantolga in https://github.com/betalgo/openai/pull/301
* sync by @kayhantolga in https://github.com/betalgo/openai/pull/309
* A couple of small fixes for the 7.1 beta by @rzubek in https://github.com/betalgo/openai/pull/302
* Dev v7.1.2 beta by @kayhantolga in https://github.com/betalgo/openai/pull/310
* Update ChatCompletionCreateRequest model comment by @WeihanLi in https://github.com/betalgo/openai/pull/332
* Some refactoring about functions by @kayhantolga in https://github.com/betalgo/openai/pull/334
* Several improvements / fixes for functions by @rzubek in https://github.com/betalgo/openai/pull/320
* Enable to set FunctionCall to AssistantRole's ChatMessage by @yt3trees in https://github.com/betalgo/openai/pull/313
* Version 7.1.3 by @kayhantolga in https://github.com/betalgo/openai/pull/337
* v7.1.3 by @kayhantolga in https://github.com/betalgo/openai/pull/341

## New Contributors
* @rzubek made their first contribution in https://github.com/betalgo/openai/pull/300
* @N0ker made their first contribution in https://github.com/betalgo/openai/pull/297
* @yt3trees made their first contribution in https://github.com/betalgo/openai/pull/313

**Full Changelog**: https://github.com/betalgo/openai/compare/v7.0.0...v7.1.3

<hr /><em>This discussion was created from the release <a href='https://github.com/betalgo/openai/releases/tag/v7.1.3'>v7.1.3</a>.</em>

**Comment by DarkLight55** [comment]
我用.net mvc 调用 CreateCompletion 为什么一直处于等待状态不返回值

**Comment by DarkLight55** [comment]
When I call CreateCompletion with.NET mvc, why do I keep waiting without returning a value
